### DISCUSSION TITLE
regridding NEMO outputs with  UGRID

### DISCUSSION THREAD

**Discussion by jupalm** [SciTools/iris#5172] [Q&A] [answered]
Hi, 

I am trying to regrid NEMO outputs (irregular grid) on a regular grid to make some nice zonal average plots with iris.
i manage to use the normal cube.regrid(reg_cube, iris.analysis.PointInCell())
Which works well almost everywhere but at high latitude, where there is a lot of missing data.

So i ended up trying to use iris-esmf-regrid which sounds like exactly what i need, all functions to regrid from unstructured  to regular grid are there.
But the first answer i got  when trying is that my loaded cube has no mesh
so i tried loading the cube as said in the doc : 
using the  
```
from iris.experimental.ugrid import PARSE_UGRID_ON_LOAD

with PARSE_UGRID_ON_LOAD.context():
            cube = iris.load_cube(file_name) 
```
but here again, iris struggles to get the mesh from the netcdf file.
i tried to use the NEMO mesh file from  the same run without much success. 

In the end, i am not sure if this UGRID functions can be used with NEMO nc files, 
Could you explain if possible, how to use these functions to read a NEMO nc files and get a cube with a mesh that the regrid functions would be happy with.
It looks like the step i miss to  get the whole thing running.

Huge thanks in advance for your help!

Best regards, 

Julien.

**Comment by trexfeathers** [comment, marked as answer]
Hi @jupalm, thanks for getting in touch!

Here are the regridders that can take **curvilinear** grids (like NEMO/ORCA) as inputs:

- [`iris.analysis.PointInCell`](https://scitools-iris.readthedocs.io/en/stable/generated/api/iris/analysis.html#iris.analysis.PointInCell) - handles anything 'unstructured', including curvilinear grids
- [`iris.analysis.UnstructuredNearest`](https://scitools-iris.readthedocs.io/en/stable/generated/api/iris/analysis.html#iris.analysis.UnstructuredNearest) - handles anything 'unstructured', including curvilinear grids
- [`esmf_regrid.schemes.ESMFAreaWeighted`](https://iris-esmf-regrid.readthedocs.io/en/stable/_api_generated/esmf_regrid.schemes.html#esmf_regrid.schemes.ESMFAreaWeighted) - handles rectilinear or curvilinear grids

A **mesh**, usually under the UGRID specification, is a different concept to a curvilinear grid - you do not need to use `PARSE_UGRID_ON_LOAD` for any of the regridders above; they will all operate on your `Cube` when loaded normally.

### Mesh handling

iris-esmf-regrid also has a mesh-specific regridder - [`esmf_regrid.experimental.unstructured_scheme.MeshToGridESMFRegridder`](https://iris-esmf-regrid.readthedocs.io/en/stable/_api_generated/esmf_regrid.experimental.unstructured_scheme.html#esmf_regrid.experimental.unstructured_scheme.MeshToGridESMFRegridder) - which may be worth trying to see if the results are better for you. To try this out, you need to first convert your ORCA grid into a UGRID mesh format - [see here](https://scitools-iris.readthedocs.io/en/stable/further_topics/ugrid/other_meshes.html#nemo-data-on-orca-tripolar-grid). Once you have a 'mesh-`Cube`', [see here](https://scitools-iris.readthedocs.io/en/stable/further_topics/ugrid/operations.html#regridding) for instructions on regridding (I think you might have seen this already).

**Reply by jupalm** [reply]
Thanks!! 
That's super useful!
I only saw the PointInCell method as the documentation on the old experimental tools only point to this one [here](https://scitools-iris.readthedocs.io/en/latest/generated/api/iris/experimental/regrid.html?highlight=regrid%20curvilinear#iris.experimental.regrid.regrid_weighted_curvilinear_to_rectilinear)
The explanations on how to convert the ORCA file are brilliant exactly what i needed. 
Might be worth doing a conversion toolbox (maybe in iris-esmf-regrid) that could be useful.

Thanks again for your help!
